includes/admin/discounts/discount-actions.php

```diff
@@ -30,8 +30,9 @@ function edd_add_discount( $data ) {
 		wp_die( __( 'You do not have permission to create discount codes', 'easy-digital-downloads' ), __( 'Error', 'easy-digital-downloads' ), array( 'response' => 403 ) );
 	}
 
-	// Setup the discount code details
-	$posted = array();
+	if( edd_get_discount_by_code( $data['code'] ) ) {
+		wp_redirect( add_query_arg( 'edd-message', 'discount_exists', $data['edd-redirect'] ) ); edd_die();
```

Comment:
Should this query arg be passed through `esc_url`?

Reply:
Wrapping the URL leads to bad URLs being generated: `edit.php?post_type=download#038;page=edd-discounts&#038;edd-message=discount_added`. This then just renders the downloads list table.